packages/block-editor/src/hooks/duotone.js

```diff
@@ -258,50 +307,26 @@ const withDuotoneStyles = createHigherOrderComponent(
 			props.name,
 			'color.__experimentalDuotone'
 		);
-		const duotonePalette = useMultiOriginPresets( {
-			presetSetting: 'color.duotone',
-			defaultSetting: 'color.defaultDuotone',
-		} );
-
-		const id = `wp-duotone-${ useInstanceId( BlockListBlock ) }`;
-
-		let colors = props?.attributes?.style?.color?.duotone;
 
-		if ( ! Array.isArray( colors ) ) {
-			const duotone = duotonePalette.find( ( dt ) => dt.slug === colors );
-
-			if ( duotone ) {
-				colors = duotone.colors;
-			}
-		}
+		const id = `wp-duotone-flibble-${ useInstanceId( BlockListBlock ) }`;
+		const className = classnames( props?.className, id );
+		const duotoneStyle = props?.attributes?.style?.color?.duotone;
 
-		if ( ! duotoneSupport || ! colors ) {
+		// CAUTION: code added before the early return will be executed
+		// for all blocks, not just those that support duotone. Code added
+		// above this line should be carefully evaluated for its impact on
+		// performance.
+		if ( ! duotoneSupport || ! duotoneStyle ) {
```

Comment:
There might be a bug here, if we render conditionally depending on whether there's a duotone style, the block will "remount" entirely when when apply a duotone style to a block that supports duotone. In other words, I'd move the condition to the React tree instead, that way the React element tree has a similar "shape" which allows React to reconcile it properly without remounting.

Reply:
Do you mean?

```jsx
{ duotoneStyle && (
	<BlockDuotoneStyles
		name={ props?.name }
		duotoneStyle={ duotoneStyle }
		id={ id }
	/>
) }
```

Reply:
Pushed a change. LMK.